python_scraper/python_scraper/spiders/sec_nport_p.py

```diff
@@ -6,25 +6,43 @@
 
 
 class SecNPORTPSpider(scrapy.Spider):
+    # Unique spider name
     name = 'sec_nport_p'
+
+    # Parse URLS as long as they're from the 'sec.report' domain
     allowed_domains = ['sec.report']
+
+    # Start requests here
     start_url = 'https://sec.report/Document/Header/?formType=NPORT-P&keyword=corporate'
-    start_pages = 8
 
+    # Parse through page
+    max_pages = 15
+
+    # Called by Scrapy when spider begins
+    # Provide Request information for the Scrapy Downloader
     def start_requests(self):
-        for page in range(1, self.start_pages+1):
+        for page in range(1, self.max_pages + 1):
             yield Request(f'{self.start_url}&page={page}', callback=self.parse)
 
+    # Lambda method passed into Request value in start_requests method
+    # Run by Scrapy Downloader
+    #
+    # Response holds nport search page information generated by Scrapy Downloader
     def parse(self, response):
+        # Parse response in XPath format
         for url in response.xpath('//table//tr/td[2]/a/@href').extract():
             yield response.follow(f'{url}primary_doc.xml', callback=self.parse_xml, headers={
                 'Accept': 'application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
                 'Accept-Language': 'en',
             })
 
+    # Lambda method added to Scrapy parsing queue
+    # Called in the parse lambda
+    #
+    # Response contains parsed XML nport info
     def parse_xml(self, response):
         response.selector.register_namespace('x', 'http://www.sec.gov/edgar/nport')
-        yield CompanyItem(
```

Comment:
Why not yield?

Reply:
Either works. The return keyword seems more intuitive to its use.

Reply:
Okie, sure we can change for this case. Maybe you know, but if you don't, you can only return one thing from a function but you can yield as many as you like. In the future we will most likely be yielding multiple items from the xml so then we would yield them.

Reply:
Yes I'm aware of the difference. When we're returning from a for loop it makes sense to yield but in this case we're not so return seems best.